.github/workflows/_build-production-node-and-runtime.yml

```diff
@@ -65,49 +65,27 @@ jobs:
           aws-region: ${{ env.AWS_REGION }}
 
       - name: Copy production binary to S3 Mainnet AWS bucket
+        uses: ./.github/actions/copy-file-to-s3
         if: >
           env.SECRETS_AWS_MAINNET_ACCESS_KEY_ID != '' &&
           env.SECRETS_AWS_MAINNET_SECRET_ACCESS_KEY != ''
-        shell: bash
-        env:
-          BINARY_DIR: target/production
-          BINARY_FILE: aleph-node
+        with:
+          binary-dir: target/production
+          binary-file: aleph-node
           # yamllint disable-line rule:line-length
-          S3BUCKET_PATH: builds/aleph-node/commits/${{ steps.get-ref-properties.outputs.sha }}/aleph-node
-          S3BUCKET_FILE: aleph-node-${{ steps.get-ref-properties.outputs.sha }}.tar.gz
-          S3BUCKET_URL: s3://${{ secrets.CI_MAINNET_S3BUCKET_NAME }}
-        run: |
-          aws s3api head-object \
-            --bucket '${{ secrets.CI_MAINNET_S3BUCKET_NAME }}' \
-            --key '${{ env.S3BUCKET_PATH }}/${{ env.S3BUCKET_FILE }}' || not_exist=true
-          if [[ "${not_exist}" ]]; then
-            tar -cvzf ${{ env.S3BUCKET_FILE }} -C ${{ env.BINARY_DIR }} ${{ env.BINARY_FILE }}
-            aws s3 cp ${{ env.S3BUCKET_FILE }} \
-              ${{ env.S3BUCKET_URL }}/${{ env.S3BUCKET_PATH }}/${{ env.S3BUCKET_FILE }}
-          else
-            echo "File ${{ env.S3BUCKET_FILE }} already exists on CI S3 bucket"
-          fi
+          s3-bucket-path: builds/aleph-node/commits/${{ steps.get-ref-properties.outputs.sha }}/aleph-node
```

Comment:
Something that not necessarily have to be done in this PR...  I was thinking if instead of passing that path in the workflow we could just have a more general action to "store artifact X from repo Y for commit Z" where you would pass X, Y and Z instead of path.  In that scenario, the action could be used in other repositories without worrying that the path is always `builds/Y/commits/Z/X`, as it's going to be concatenated by the action.
But then, this action would have to be renamed to `store-commit-artifact-in-s3` - or it can be a separate action re-using `copy-to-s3`.

Reply:
Agree it would be value added, yet we don't have any more usecases to actually be covered by that new funcionality.